internal/kernel/kernel.go

```diff
@@ -0,0 +1,15 @@
+package kernel
+
+import "strings"
+
+func replaceInvalidChar(r rune) rune {
+	if (r < 'A' || r > 'Z') && (r < 'a' || r > 'z') && (r < '0' || r > '9') && r != '-' && r != '.' && r != '_' {
+		return '_'
+	}
```

Comment:
Shouldn't we just replace `+` with `_`?

Reply:
Handling any invalid character makes the code more future-proof than just handling `+`.